src/Distribution/Server/Features/UserNotify.hs

```diff
@@ -357,12 +453,109 @@ initUserNotifyFeature env@ServerEnv{ serverStateDir, serverTemplatesDir,
                    [serverTemplatesDir, serverTemplatesDir </> "UserNotify"]
                    [ "user-notify-form.html" ]
 
-    return $ \users core uploadfeature adminlog userdetails reports tags -> do
+    return $ \users core uploadfeature adminlog userdetails reports tags revers -> do
       let feature = userNotifyFeature env
                       users core uploadfeature adminlog userdetails reports tags
-                      notifyState templates
+                      revers notifyState templates
       return feature
 
+-- | Get the release notification emails when a new package has been released.
+--   The new package (PackageIdentifier) must already be in the indexes.
+dependencyReleaseEmails
+  :: forall m. Monad m
+  => (PackageName -> m UserIdSet)
+  -> PackageIndex.PackageIndex PkgInfo
+  -> ReverseIndex
+  -> (UserId -> Maybe UserName)
+  -> URI
+  -> (UserId -> m (Maybe NotifyPref))
+  -> UTCTime
+  -> PackageIdentifier
+  -> m (Map.Map UserId [String])
+dependencyReleaseEmails userSetIdForPackage index (ReverseIndex revs nodemap dependencies) idToName baseURI queryGetUserNotifyPref now pkgId =
+  case lookup (pkgName pkgId) nodemap :: Maybe NodeId of
+    Nothing -> pure mempty
+    Just foundPackage -> do
+      let
+        vertices :: Set.Set Vertex
+        vertices = suc revs foundPackage
+        revDepNames :: [PackageName]
+        revDepNames = mapMaybe (`lookupR` nodemap) (Set.toList vertices)
+      toNotify <- traverse maintainersToNotify revDepNames
+      pure $
+        Map.fromList
+          [ ( maintainerId
```

Comment:
What happens here if multiple packages by the same maintainer are affected by the same release? Ideally, I think the maintainer should get a single notification with a list of those packages.

Reply:
Ok, I have pushed a commit which groups by userId and packageId. Now the email looks like this:

```
The dependency transformers-0.6.0.2 has been updated.

You have requested to be notified when a dependency isn't accepted by any of your maintained packages.

These are your packages that require transformers but don't accept 0.6.0.2:

servant-0.19.1

You can adjust your notification preferences at

http://gorm.local:8080/user/admin/notify
```